### PR TITLE
ZoteroHarvesterConfig: validation

### PR DESCRIPTION
**Pull request by mtrojan-ub**

### CODE REVIEW

File: cpp/lib/src/ZoteroHarvesterConfig.cc [hunk 1]
```diff
@@ -143,49 +180,60 @@ GlobalParams::GlobalParams(const IniFile::Section &config_section) {
...
     name_ = group_section.getSectionName();
     user_agent_ = group_section.getString(GetIniKeyString(USER_AGENT));
     isil_ = group_section.getString(GetIniKeyString(ISIL));
     output_folder_ = group_section.getString(GetIniKeyString(OUTPUT_FOLDER));
     author_swb_lookup_url_ = group_section.getString(GetIniKeyString(AUTHOR_SWB_LOOKUP_URL));
     author_lobid_lookup_query_params_ = group_section.getString(GetIniKeyString(AUTHOR_LOBID_LOOKUP_QUERY_PARAMS), "");
-}
 
+    // Warnings for unknown entries
+    for (const auto &entry : group_section) {
```
Comment (ruschein) [inline cpp/lib/src/ZoteroHarvesterConfig.cc:221]: This looks exactly like some other new code further up.  Please encapsulate both in a new function.

Comment (mtrojan-ub) [reply cpp/lib/src/ZoteroHarvesterConfig.cc:221]: There are 3 validation mechanisms looking similar but they are not exactly the same (additional functions are called for validation and the error messages differ as well)

Comment (mtrojan-ub) [reply cpp/lib/src/ZoteroHarvesterConfig.cc:221]: i see what i can do

Comment (mtrojan-ub) [reply cpp/lib/src/ZoteroHarvesterConfig.cc:221]: I could only achieve that with a template function. How do you define a generic parameter for a map of any enum to a std::string?

Comment (ruschein) [reply cpp/lib/src/ZoteroHarvesterConfig.cc:221]: Yes, but it is a trivial template.

File: cpp/lib/include/ZoteroHarvesterConfig.h [hunk 2]
```diff
@@ -59,8 +60,44 @@ struct MarcMetadataParams {
 public:
     MarcMetadataParams() = default;
     MarcMetadataParams(const IniFile::Section &config_section);
+    static bool IsValidIniEntry(const IniFile::Entry &entry);
 };
```
Comment (ruschein) [inline cpp/lib/include/ZoteroHarvesterConfig.h:65]: Missing blank line.

File: cpp/lib/include/ZoteroHarvesterConfig.h [hunk 3]
```diff
@@ -59,8 +60,44 @@ struct MarcMetadataParams {
...
+        bool valid(false);
+        for (const auto &allowed_value : allowed_values) {
+            if (entry.name_ == allowed_value.second) {
+                valid = true;
+                break;
+            }
+        }
+
+        if (not valid) {
+            for (unsigned i(0); i<callbacks.size(); ++i) {
```
Comment (ruschein) [inline cpp/lib/include/ZoteroHarvesterConfig.h:84]: You can go back to the for-in loop.  Also, this version is missing spaces.

**Review by ruschein — CHANGES_REQUESTED** [review]

**Pull request merged**